java-checks/src/main/java/org/sonar/java/checks/helpers/ExpressionsHelper.java

```diff
@@ -235,4 +237,19 @@ public static boolean alwaysReturnSameValue(ExpressionTree expression) {
     return true;
   }
 
+  public static Optional<Symbol> getInvokedSymbol(MethodInvocationTree mit) {
+    ExpressionTree methodSelect = mit.methodSelect();
+    if (methodSelect.is(Tree.Kind.MEMBER_SELECT)) {
+      ExpressionTree expression = ((MemberSelectExpressionTree) methodSelect).expression();
+      if (expression.is(Tree.Kind.IDENTIFIER)) {
+        return Optional.of(((IdentifierTree) expression).symbol());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static boolean isNotReassigned(Symbol symbol) {
+    return symbol.isFinal() || (symbol.isVariableSymbol() && JUtils.isEffectivelyFinal(((Symbol.VariableSymbol) symbol)));
+  }
```

Comment:
Good idea to extract this method. You can also extract it in `OSCommandsPathCheck.java`, using the same code.